pages/user-interfaces.html

```diff
@@ -0,0 +1,160 @@
+can be extended by the minor Coq
+mode <a href="https://github.com/cpitclaudel/company-coq">Company-Coq</a>.
+These are widely used.  In particular, Proof General was one of the
+first user interface for proof assistants in general, and for Coq in
+particular.  Proof General also has an
+experimental <a href="https://github.com/ProofGeneral/PG/tree/async">async
+branch</a> that relies
+on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
+XML protocol for IDEs</a>, and supports new features such as
+asynchronous proof processing.
+</li>
+
+<li>
+<strong>Vim</strong> users can use the actively
+maintained <a href="https://github.com/whonore/Coqtail">Coqtail</a>
+extension.  This extension relies
+on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
+XML protocol for IDEs</a>.  It is partly based upon a previous, and
+now unmaintained, Vim extension
+called <a href="https://github.com/the-lambda-church/coquille">Coquille</a>.
+</li>
+
+<li>
+<strong>VsCode</strong> users can use
+the <a href="https://github.com/coq-community/vscoq">VsCoq
+extension</a> that is actively maintained as part
+of <a href="https://github.com/coq-community/manifesto">coq-community</a>.
+This extension indirectly relies
+on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
+XML protocol for IDEs</a>, until Coq provides a
+native <a href="https://en.wikipedia.org/wiki/Language_Server_Protocol">Language
+Server Protocol</a>.  VsCoq was initially authored by C.J. Bell, but
+was left unmaintained for several years before being forked in
+coq-community.
+</li>
+
+</ul>
+
+</p>
+
+<p>
+An experimental alternative to the editors mentioned above is the
+computational notebook interface provided by Jupyter.
+A <a href="https://github.com/EugeneLoy/coq_jupyter/">Jupyter kernel
+for Coq</a> is available if you want to try this interface, for
+instance for pedagogical or readability purposes.  One of the
+advantages of using a Jupyter notebook is that you can save and
+display a selection of intermediate proof steps, which your reader
+will see without the need to reexecute the document.  This kernel
+relies
+on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
+XML protocol for IDEs</a>.
+</p>
+
+</div>
+
+<div class="frameworklinks">
+<ul>
+<li><a class="extlink" href="https://proofgeneral.github.io/">Proof General (Emacs)</a></li>
+<li><a class="extlink" href="https://github.com/whonore/Coqtail">Coqtail (Vim)</a></li>
+<li><a class="extlink" href="https://github.com/coq-community/vscoq">VsCoq (VsCode)</a></li>
+<li><a class="extlink" href="https://github.com/EugeneLoy/coq_jupyter">Jupyter kernel</a></li>
+</ul>
+</div>
+</div>
+
+<div class="framework">
+<div class="frameworklabel">Standalone interfaces</div>
+<div class="frameworkcontent">
+
+<p>
+If you are not a user of any of the previously mentioned editors, you
```

Comment:
> like supporting more of the features enabled by the XML protocol

which ones?

Reply:
The ones that are listed. But yeah, you say it's not just CoqIDE. What about proof diffs?

Reply:
That's even worse. It was supported by VsCoq even before an argument was made that it was impossible/hard do on the UI side, and we added it to Coq itself...

Reply:
Well sure, PG also had some diff support. Though, the interest of providing such features as part of Coq itself (in theory at least) is that it makes them available at a lower cost on all interfaces (no need to reimplement it several times). Kind of similar to the ideas behind LSP. It would seem that diffs are of interest at least to some: https://github.com/whonore/Coqtail/issues/57